test/common/TestBedChild.cpp

```diff
@@ -659,12 +668,48 @@ namespace RcclUnitTesting
     }
 
     // Synchronize
-    for (int localRank : localRanksToExecute)
+    if (timeout)
     {
-      if (this->verbose) INFO("Starting synchronization for rank %d\n", localRank);
-      CHECK_HIP(hipSetDevice(this->deviceIds[localRank]));
-      for (int i = 0; i < this->numStreamsPerGroup; i++)
-        CHECK_HIP(hipStreamSynchronize(this->streams[localRank][i]));
+      std::atomic<bool> timesUp{false};
+      int totalStreams = localRanksToExecute.size() * this->numStreamsPerGroup;
+      bool streamsComplete;
+      if (this->verbose) INFO("Starting sychronization and timing\n");
+      std::thread timer(wait, timeout, std::ref(timesUp));
```

Comment:
This looks overly complicated.
You already have a thread that is looping to check for status completion, so it's hard to justify adding an additional thread to just check for timing status.

Your loop can easily be:
- Get start time
- Loop over remaining uncompleted streams and check for completion.
- If not all done yet, get the current time, and check if currtime - starttime > timeout.  If so, then issue aborts on remaining comms, then return timeout status.